src/codewhisperer/service/recommendationHandler.ts

```diff
@@ -60,6 +60,10 @@ const nextCommand = Commands.declare('editor.action.inlineSuggest.showNext', ()
 })
 
 const rejectCommand = Commands.declare('aws.codeWhisperer.rejectCodeSuggestion', () => async () => {
+    if (!AuthUtil.instance.isConnectionValid()) {
```

Comment:
@Will-ShaoHua sorry for making you change again. But I just want to verify if we want to use `isConnectionValid()` or `isConnected()`. I see when we assign `CODEWHISPERER_ENABLED` a value we use `isConnected()` which means the connection exists but isn't guaranteed to work.

I'm not sure what is expected, but I think that you may want to execute the logic even if the connection is invalid. Then if that fails due to invalid connection we will prompt user to reauth.

Reply:
yea that concern is understandable, I use simple `return ` here unlike doing another expiration check then prompt reauth is because `reject` command, theoretically(if our code is working correctly) will only be registered when there is a valid connection (usable one)

Reply:
tbh.. authUtil.ts is another CW tech debt to deal with

Reply:
100%, this applies to CodeCatalyst as well. I'd be interested to work on it at some point.